python/pyspark/sql/connect/dataframe.py

```diff
@@ -201,6 +202,34 @@ def sort(self, *cols: "ColumnOrString") -> "DataFrame":
         """Sort by a specific column"""
         return DataFrame.withPlan(plan.Sort(self._plan, *cols), session=self._session)
 
+    def sample(
+        self,
+        withReplacement: Optional[Union[float, bool]] = None,
+        fraction: Optional[Union[int, float]] = None,
+        seed: Optional[int] = None,
```

Comment:
yes, that's a bit confusing at first glance.

Reply:
Yes, if we can break the signature, it would be:

```py
def sample(
    self,
    fraction: float,
    *,
    withReplacement: Optional[bool] = None,
    seed: Optional[int] = None,
) -> "DataFrame":
    ...
```

Reply:
`withReplacement` can be `: bool = False` if the default is `False`.

Reply:
I like this idea